### PR TITLE
Show "Offline Compile" for offline compile only jobs

### PR DESCRIPTION
**Pull request by rwalker777**
# What does this implement/fix?

Currently an offline device compile job shows "Install", change to "Offline Compile" for more accurate label in firmware job display window.

**Related issue or feature (if applicable):**

- fixes <link to issue>

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue) — `bugfix`
- [ ] New feature (non-breaking change which adds functionality) — `new-feature`
- [ ] Enhancement to an existing feature — `enhancement`
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) — `breaking-change`
- [ ] Refactor (no behaviour change) — `refactor`
- [ ] Documentation only — `docs`
- [ ] Maintenance / chore — `maintenance`
- [ ] CI / workflow change — `ci`
- [ ] Dependencies bump — `dependencies`

## Checklist

- [x] The code change is tested and works locally.
- [x] `npm run lint` passes.
- [x] `npm run test` passes.
- [ ] Tests have been added to verify that the new code works (where applicable).

### CODE REVIEW

**Review by esphbot — CHANGES_REQUESTED** [review]
Blocking issues found.

- `firmwareJobTypeLabel` has no callers — the PR's stated fix is not delivered
- New doc comment describes behavior that does not exist

File: src/components/shared/firmware-jobs-list.ts [hunk 1]
```diff
@@ -105,7 +108,7 @@ function renderJob(host: FirmwareJobsListHost, job: FirmwareJob): TemplateResult
   const name = host._jobDisplayName(job);
   const effectiveType = effectiveJobType(job);
   const typeIcon = TYPE_ICONS[effectiveType] ?? "hammer-wrench";
-  const typeLabel = host._localize(`firmware_jobs.type_${effectiveType}`);
+  const typeLabel = firmwareJobTypeLabel(job, host._localize);
```
[the remark this answers is not in the document]
Comment (rwalker777) [reply src/components/shared/firmware-jobs-list.ts]: This is done.

[the remark this answers is not in the document]
Comment (esphbot) [reply src/components/shared/firmware-jobs-list.ts]: Confirmed fixed. Icon now derives from `job.job_type`, so deferred installs render `hammer-wrench` beside "Offline compile".

Verified no regression elsewhere: `effectiveJobType` returned `job_type` unchanged for every non-deferred job, so only deferred installs shift.

One downstream note — `deriveFollowCommandType` (`commands.ts:32`) still maps deferred installs to `"install"`, titling the reopened dialog "Install — {name}". Row and dialog now disagree. Details in review.

**Review by esphbot — CHANGES_REQUESTED** [review]
Blocking issues found.

- List row now says "Offline compile" but clicking it opens a dialog titled "Install"

**Review by esphbot — CHANGES_REQUESTED** [review]
Blocking issues found.

- Retry on an offline_compile dialog wipes the log and hangs on a spinner forever
- Missing `command.offline_compile_failed` — a failed offline compile renders the raw translation key

**Review by esphbot — CHANGES_REQUESTED** [review]
> [!WARNING]
> Important issues found.
>
> - A failed offline compile loses the clean / reset build-failure recovery links
> - Run timer disappears for offline compiles — contradicting the docblock that calls out this exact case

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

File: src/util/firmware-job-display.ts [hunk 2]
```diff
@@ -9,11 +9,40 @@ import type { LocalizeFunc } from "../common/localize.js";
  *
  * Deliberately NOT used by the device card's busy badge — "Compiling"
  * during a deferred install matches a normal chain's compile phase.
+ * Also NOT used for the job-type text in the firmware-tasks list — see
+ * ``firmwareJobTypeLabel`` below, which surfaces deferred installs as
+ * "Offline compile" instead, since the device is offline and nothing
+ * is actually being installed yet.
```
[the remark this answers is not in the document]
Comment (bdraco) [reply src/util/firmware-job-display.ts]: Fixed on the branch: `renderers.ts` now imports and calls `firmwareJobTypeLabel(job, host._localize)` for the row label, so the deferred-install case renders "Offline compile".

File: src/util/firmware-job-display.ts [hunk 3]
```diff
@@ -9,11 +9,40 @@ import type { LocalizeFunc } from "../common/localize.js";
...
+ * is offline, so nothing installs yet — but "Compile" alone loses the
+ * context that an install is queued behind it. This surfaces it as
+ * "Offline compile" instead of letting it read as an outright Install.
+ */
+export function firmwareJobTypeLabel(job: FirmwareJob, localize: LocalizeFunc): string {
+  if (job.is_deferred_install) {
+    return localize("firmware_jobs.type_offline_compile");
+  }
+  return localize(`firmware_jobs.${JOB_TYPE_LABEL_KEYS[job.job_type]}`);
+}
```
[the remark this answers is not in the document]
Comment (bdraco) [reply src/util/firmware-job-display.ts]: Fixed on the branch: the `JOB_TYPE_LABEL_KEYS` map was removed. `firmwareJobTypeLabel` now uses the `type_${job.job_type}` key convention directly, with only the deferred-install branch special-cased.

[the remark this answers is not in the document]
Comment (bdraco) [reply src/util/firmware-job-display.ts:24]: Covered: the component tests in `test/components/shared/firmware-jobs-list.test.ts` exercise all three label branches through `renderGroups` (deferred install renders `type_offline_compile`, plain compile stays `type_compile`, standard install stays `type_install`), and `translation-key-coverage.test.ts` pins the new key in en.json.

**Comment by bdraco** [comment]
Pushed 5cf18cf3 (maintainer edit) to close an edge case found in local review: the backend also sets is_deferred_install on a failed OTA upload it converts offline (the mid-flash disconnect shape), so that row read "Offline compile" and reopened as an offline-compile dialog even though what ran was a flash of a finished build. Both call sites now gate the special case on the job being a COMPILE; the converted upload keeps its honest Upload label and reopens as the Install it was. Tests pin both shapes, plus the plain-compile derivation the earlier rewrite dropped.

@esphbot review

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Comment by bdraco** [comment]
Good catch, taken in 142c3c50: the gate now includes offline_compile, so a reopened remote deferred install keeps its Build locally link (the resubmit path already handles the offline shape; firmwareInstall just re-defers force-local and the dialog stays attached). Two sub-line tests pin the link shown for offline_compile and still hidden for a plain compile.

@esphbot review

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by bdraco — APPROVED** [review]

**Comment by bdraco** [comment]
Thanks @rwalker777
